src-model/func_for_model_creation.R

```diff
@@ -0,0 +1,115 @@
+#' all the functions required for fitting models
+
+require(tidyverse)
+require(vcdExtra)
+require(rstan)
+require(splines)
+#' Create a stan model for a given target date and dataset
+#'
+#' @param data is a dataframe containing the columns observations, location and date.
+#' @param stan_file is the path to which stan file to use
+#' @param num_seq is the number of seq required in the last 60 days to be modeled, a numeric integer
+#' @param target_date target date is the last day of the data to use, needs to be a date object
+#' @param num_days is the number of days before the target date to use, a numeric integer
+#' @param target_loc allows you to choose what locations to use, a string vector, default is NULL
+#' @param iterations,warmup control the number of iterations and warmup iterations stan uses
+#' @return return a list containing the stan object, the number of locations, the number of clades, and the locations
+stan_maker <- function(data,stan_file, num_seq = 1, target_date = Sys.Date(), num_days = 150, target_loc = NULL, interations = 3000, warmup = 1000){
```

Comment:
minor stylistic comment: When you have a lot of arguments, I suggest putting them on separate lines, like
```
stan_maker <- function(data,
                                         stan_file, 
                                         num_seq = 1, ...
```
for better readability